src/ray/core_worker/core_worker.cc

```diff
@@ -2103,6 +2088,15 @@ Status CoreWorker::Wait(const std::vector<ObjectID> &ids,
           worker_context_,
           &ready));
     }
+  } else {
+    // When we don't need to fetch_local, we can simply just add the object_ids that we
+    // know are ready in plasma until we have num_objects.
```

Comment:
This comment is a bit misleading. We don't know whether or not the objects are ready in the local plasma store, just that the object exists somewhere on the cluster. Suggest:
```suggestion
    // When we don't need to fetch_local, we don't need to wait for the objects to be pulled to
    // the local object store, so we can directly add them to the ready set.
```